test/jellyfish_web/socket_test.exs

```diff
@@ -1,102 +1,121 @@
 defmodule JellyfishWeb.SocketTest do
-  use ExUnit.Case
+  use JellyfishWeb.ConnCase
 
   import ExUnit.CaptureLog
 
-  alias Jellyfish.Peer.WebRTC
-  alias Jellyfish.{Room, RoomService}
-
-  alias JellyfishWeb.{Endpoint, Socket}
+  alias Jellyfish.RoomService
+  alias JellyfishWeb.{PeerToken, Socket}
 
   @data "mediaEventData"
 
-  setup do
-    {:ok, room} = RoomService.create_room(10)
-    {:ok, room_pid} = RoomService.find_room(room.id)
-    {:ok, peer} = Room.add_peer(room_pid, WebRTC)
+  setup %{conn: conn} do
+    room_conn = post(conn, ~p"/room", maxPeers: 1)
+    assert %{"id" => room_id} = json_response(room_conn, :created)["data"]
+    {:ok, room_pid} = RoomService.find_room(room_id)
+
+    conn = post(conn, ~p"/room/#{room_id}/peer", type: "webrtc")
+
+    assert %{"token" => token} = json_response(conn, :created)["data"]
 
     on_exit(fn ->
-      RoomService.delete_room(room.id)
+      room_conn = delete(conn, ~p"/room/#{room_id}")
+      assert response(room_conn, :no_content)
     end)
 
-    %{room_id: room.id, peer_id: peer.id, room_pid: room_pid}
+    {:ok,
+     %{
+       room_pid: room_pid,
+       authenticated?: false,
+       token: token
+     }}
   end
 
   describe "connecting" do
-    test "when credentials are valid", %{room_id: room_id, peer_id: peer_id, room_pid: room_pid} do
-      assert {:ok, state} = connect(%{"room_id" => room_id, "peer_id" => peer_id})
-      assert state.peer_id == peer_id
-      assert state.room_pid == room_pid
-
-      status =
-        room_pid
-        |> Room.get_state()
-        |> Map.get(:peers)
-        |> Map.get(peer_id)
-        |> Map.get(:status)
-
-      assert status == :connected
+    test "when credentials are valid" do
+      assert {:ok, _state} = connect(%{})
```

Comment:
I think this test is kinda useless now, at least it doesn't match the name `the credentials are valid` when there's no credentials anywhere.